File: backend/models/support.py
from typing import Optional

from pydantic import BaseModel


class SupportMetricsResponse(BaseModel):
    total_tickets: int
    open_tickets: int
    high_priority_tickets: int
    sla_breaches: int
    avg_resolution_time_hours: Optional[float] = None
    avg_satisfaction_rating: Optional[float] = None

File: frontend/src/types/support.ts
/** Matches backend/models/support.py::SupportMetricsResponse */
export interface SupportMetricsResponse {
  total_tickets: number
  open_tickets: number
  high_priority_tickets: number
  sla_breaches: number
  avg_resolution_time_hours: number | null
  avg_satisfaction_rating: number | null
}
